YvesIraguha/lost-and-found-api
# #171926667 setting profile picture

#### What does this PR do?
uploading profile picture
#### Description of Task to be completed?
- setting up multer for handling multi-form data
- saving a file using multer
- updating users photoUrl using file destination
#### How should this be manually tested?
- clone the repo `git clone`
- checkout the branch `git checkout `ft-create-profile-#171926667`
- install all dependencies `npm install`
- start the server `npm run dev`
- using postman sign up or login to get *auth-token* on `http://localhost:3000/api/v1/users/signUp`
- in postman on `http://localhost:3000/api/v1/users/profile/image` set header to *auth-token* and insert its value
- on the same end-point, in body check form-data radio instead of raw
- set *key* select *file* instead of Text and call it *profilePicture* then select any picture you want to upload then *Send your request*
> You should see a seccessful message and updated user
-  by copying the returned photoUrl and paste it in brower after `http://localhost:3000` *eg: `http:localhost:3000/uploads//myProfilePicture.jpg`  
> you should see the uploaded picture
#### Any background context you want to provide?
- Image to upload should be of jpeg/jpg or png file exptension only
- You don't have to set other custom headers like content-type except *auth-token*
#### What are the relevant pivotal tracker stories?
#171926667
